packages/core/src/http-handler.ts

```diff
@@ -28,30 +28,18 @@ export class HttpHandler {
     this._routes.registerController(name, spec);
   }
 
-  protected _handleRequest(request: ServerRequest, response: ServerResponse): Promise<void> {
+  protected async _handleRequest(request: ServerRequest, response: ServerResponse): Promise<void> {
     const parsedRequest: ParsedRequest = parseRequestUrl(request);
     const requestContext = this._createRequestContext(request, response);
 
-    // TODO(bajtos) bind findRoute to requestContext
-    const findRoute: FindRoute = (req) => {
-      const found = this._routes.find(req);
-      if (!found) {
-        throw new HttpErrors.NotFound(
-          `Endpoint "${req.method} ${req.path}" not found.`);
-      }
-      this._bindRouteInfo(requestContext, found.controller, found.methodName);
-      return found;
-    };
-
-    // TODO(bajtos) bind invoke to requestContext
-    const invoke: InvokeMethod = async (controllerName, method, args) => {
-      const controller: { [opName: string]: Function } = await requestContext.get(controllerName);
-      const result = await controller[method](...args);
-      return result;
-    };
+    this._bindFindRoute(requestContext);
+    this._bindInvokeMethod(requestContext);
+
+    const findRoute = await requestContext.get('findRoute');
+    const invokeMethod = await requestContext.get('invokeMethod');
```

Comment:
Wanted to get your opinion as to whether we should move all the above lines to the application level. I was thinking along the lines of:

```
// application.ts
protected _handleHttpRequest(request: ServerRequest, response: ServerResponse) {
    this._setupHandlerIfNeeded();
    const parsedRequest: ParsedRequest = parseRequestUrl(request);
    const requestContext = this._createRequestContext(request, response);
    bindFindRoute(requestContext);
    bindInvokeMethod(requestContext);
    return this._httpHandler.handleRequest(requestContext);
}
```